apps/explorer/app/contract/[id]/page.tsx

```diff
@@ -42,44 +38,58 @@ export default async function Page({ params }) {
     ),
   ])
 
-  if (error) {
-    throw error
-  }
+  if (rateError) throw rateError
+  if (!rate) return notFound()
 
-  const contract = c?.contract
+  // Grab the contract and previous revisions data.
+  const [
+    [contract, contractError],
+    [previousRevisions, previousRevisionsError],
+  ] = await Promise.all([
+    to(explored.contractByID({ params: { id } })),
+    to<
+      FileContractElement[] // Not sure why this is necessary here.
+    >(explored.contractRevisions({ params: { id } })),
+  ])
 
-  if (!contract) {
-    return notFound()
-  }
+  if (contractError) throw contractError
+  if (previousRevisionsError) throw previousRevisionsError
+  if (!contract || !previousRevisions) return notFound()
 
-  const formationTxnId = getFormationTxnId(contract)
-  const finalRevisionTxnId = contract?.transaction_id || ''
+  // What should we do here? What's going on now is wrong, I'm sure, and
```

Comment:
The formation txn is the confirmation txn ID for the first ever revision.
The final revision txn ID is the confirmation txn ID for the last revision.
You already have the list of revisions, sort them by revision number.